app/main.py

```diff
@@ -1,3 +1,52 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as f:
+        data = json.load(f)
+        fuel_price = data["FUEL_PRICE"]
+        for customer_data in data["customers"]:
+            customer = Customer(**customer_data)
+            print(f"{customer.name} has {customer.money} dollars")
+            closest = {}
+            shop_names = {}
+            for shop_data in data["shops"]:
+                shop = Shop(**shop_data)
+                fuel = customer.car["fuel_consumption"]
+                customer.product_total(shop.products)
+                customer.fuel_expenses(fuel, shop.location, fuel_price)
+                expenses = (
+                    customer.product_total(shop.products)
+                    + customer.fuel_expenses(fuel, shop.location, fuel_price)
```

Comment:
You are reaping your self

Reply:
I don`t catch what you mean in both cases.

Reply:
Please clarify